### ISSUE TITLE
`atDist` constraint with Line and Text seems to do nothing

### ISSUE THREAD

**Issue by christianp** [penrose/penrose#827]
I'm following [Tutorial 3](https://penrose.gitbook.io/penrose/tutorial-3). 

For displaying the vectors' labels, there's an `atDist` constraint to put the label text 5 pixels from the line. The generated diagram almost never has the label anywhere near the line.

Here's my program (I've removed the axes to make it simpler):

`linearAlgebra.dsl`:

```
type VectorSpace
type Vector
type Scalar
predicate In(Vector, VectorSpace V)
```

`vector.sty`:

```
canvas {
  width = 800
  height = 700
}

const { 
  scalar vectorSpaceSize = 600.0
  scalar arrowheadSize = 0.7
  scalar lineThickness = 1.
  scalar arrowThickness = 1.5
  color gray = rgba(0.6, 0.6, 0.6, 1.)
  color lightBlue = rgba(0.2, 0.4, 0.8, 1.0)
  color lightGray = rgba(0.95, 0.95, 0.95, 0.015)
  color none = rgba(0., 0., 0., 0.)
}

forall VectorSpace U { 
    scalar axisSize = const.vectorSpaceSize / 2.0 
    vec2 U.origin = (0., 0.)
    vec2 o = U.origin /* just so we don't need to type U.origin everytime */
    U.axisColor = const.gray

    U.background = Rectangle {
        center : U.origin
        width: const.vectorSpaceSize
        height: const.vectorSpaceSize
        side : const.vectorSpaceSize
        fillColor : const.lightGray
        strokeColor : const.none
    }

}

forall Vector u; VectorSpace U
where In(u, U) {
    u.vector = (?, ?)
    u.shape = Line {
        start : U.origin
        end : U.origin + u.vector
        endArrowhead : true
        strokeWidth : 3.0
        strokeColor : const.lightBlue
        arrowheadSize : const.arrowheadSize
    }

    u.text = Text {
        string : u.label
        fillColor : u.shape.strokeColor
    }
    ensure contains(U.background, u.shape)
    ensure contains(U.background, u.text)
    ensure atDist(u.shape, u.text, 5.0)
    ensure minSize(u.shape)
}
```

`vector.sub`:

```
VectorSpace U
Vector v 
In(v, U)
AutoLabel All
```

Here's a representative output:

![Screenshot 2022-01-12 at 13-05-04 Penrose](https://user-images.githubusercontent.com/19513/149146742-e8109664-2cbd-4e23-855d-fde9f7f8ad9a.png)

I'd expect to see the "v" within a few pixels of the line.

It looks like the position of the label is completely random. A massive `padding` parameter to `atDist` seems to make it impossible to converge, while 0 or negative values do nothing different to small positive values.

This might just be something that has gone wrong since the tutorial was written: the tutorial refers to an `Arrow` shape, which seems not to exist any more, so I replaced it with a `Line`. Nevertheless, it should work!

Looking at the source, it looks like `atDist` is effectively a synonym for `tangentTo` at the moment. Is that right? And that in turn uses `overlapping`, which doesn't have a case for an `isRectlike` shape and a `Line`, but contains #652.

 So have I just caught you in the middle of rewriting this code?

**Comment by jiriminarcik** [comment]
`So have I just caught you in the middle of rewriting this code?` - Yes! :)

We are currently implementing a new way of handling constraints using [Minkowski penalties](https://github.com/penrose/penrose/files/7856983/minkowski_progress.pdf). Parts of the new constraints were added last week in #648, #751 and #791. I'm currently going through different edge cases and adding more shape-shape combinations in #810. `isRectlike` and `Line` will be handled using the general approach for polygons.

Thanks for a valuable feedback!

**Comment by jiriminarcik** [comment]
Fix by #810. Thanks for the feedback!
